addons/website/models/website.py

```diff
@@ -449,14 +449,24 @@ def get_url_localized(router, lang):
     # ----------------------------------------------------------
 
     @api.model
-    def get_current_website(self, fallback=True):
+    def get_current_website(self, fallback=True, only_explicit=False):
+        """
+            Return the current website.
+
+            If ``only_explicit`` is set, do not try to guess,
+            but only return the website if it is explicitly defined
+            in the session (force_website_id) or in the context (website_id).
+        """
         if request and request.session.get('force_website_id'):
             return self.browse(request.session['force_website_id'])
 
         website_id = self.env.context.get('website_id')
         if website_id:
             return self.browse(website_id)
 
+        if only_explicit:
+            return False
```

Comment:
Maybe the solution is to return the first website in this case

Reply:
I removed only_explicit and used "if request" to differentiate with -u. Much better now